spec/rubocop/cop/rspec/repeated_include_example_spec.rb

```diff
@@ -0,0 +1,233 @@
+              %{include_method} 'an x', 'other'
+              %{include_method} 'an x', 'y'
+              ^{include_method}^^^^^^^^^^^^ Repeated include of shared_examples 'an x' on line(s) [2]
+            end
+          RUBY
+          expect_no_corrections
+        end
+
+        it "registers an offense for repeated #{include_method} " \
+          'with parentheses' do
+          expect_offense(<<~RUBY, include_method: include_method)
+            describe 'foo' do
+              %{include_method}('an x', 'y')
+              ^{include_method}^^^^^^^^^^^^^ Repeated include of shared_examples 'an x' on line(s) [3]
+              %{include_method}('an x', 'y')
+              ^{include_method}^^^^^^^^^^^^^ Repeated include of shared_examples 'an x' on line(s) [2]
+            end
+          RUBY
+          expect_no_corrections
+        end
+
+        it "registers an offense for repeated #{include_method} " \
+           'with composite literal / const parameters' do
+          expect_offense(<<~RUBY, include_method: include_method)
+            describe 'foo' do
+              %{include_method}('something', [1, 'b', :c, D, { e: :f }, nil])
+              ^{include_method}^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Repeated include of shared_examples 'something' on line(s) [3]
+              %{include_method}('something', [1, 'b', :c, D, { e: :f }, nil])
+              ^{include_method}^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Repeated include of shared_examples 'something' on line(s) [2]
+            end
+          RUBY
+        end
+
+        it "accepts repeated #{include_method} with different parameters" do
+          expect_no_offenses(<<~RUBY)
+            describe 'foo' do
+              #{include_method} 'an x', 'y'
+              #{include_method} 'an x', 'z'
+              #{include_method} 'an x'
+              #{include_method} 'an x', 'y', 'z'
+            end
+          RUBY
+        end
+
+        it "accepts repeated #{include_method} with different parameters " \
+          'with parentheses' do
+          expect_no_offenses(<<~RUBY)
+            describe 'foo' do
+              #{include_method}('an x', 'y')
+              #{include_method}('an x', 'z')
+              #{include_method}('an x')
+              #{include_method}('an x', 'y', 'z')
+            end
+          RUBY
+        end
+
+        it "allows repeated #{include_method} with variable arguments" do
+          expect_no_offenses(<<~RUBY)
+            describe 'foo' do
+              #{include_method} 'one', foo
+              #{include_method} 'one', foo
+
+              #{include_method} 'two', @foo
+              #{include_method} 'two', @foo
+
+              #{include_method} 'two', @@foo
+              #{include_method} 'two', @@foo
+
+              #{include_method} 'two', $foo
+              #{include_method} 'two', $foo
+            end
+          RUBY
+        end
+
+        it "allows repeated #{include_method} with block" do
```

Comment:
I can think of examples when the block is the same, but it doesn't mean those two included examples will be the same:
```ruby
it_behaves_like 'a foo' do
  let(:x) { rand }
end
```
or
```ruby
it_behaves_like 'a global counter' do
  let(:x) { $x += 1 }
end
```

But most probably this will never be found in the wild, and if it is, it's a rare exception. More likely such occurrences will be the result of an unintentional copy-paste.

And we're in a good position to be able to detect block identity by having its AST, that case where static analysis shine.

Still, if you don't feel like implementing it here, please feel free to leave this for future improvement.

Reply:
My bad, I forgot to update the PR description. I think it makes sense to focus just on literal arguments for now without checking blocks.